app/src/org/commcare/activities/GeoPointActivity.java

```diff
@@ -47,9 +71,7 @@ protected void onCreate(Bundle savedInstanceState) {
         setTitle(StringUtils.getStringRobust(this, R.string.application_name) +
                 " > " + StringUtils.getStringRobust(this, R.string.get_location));
 
-        locationManager = (LocationManager)getSystemService(Context.LOCATION_SERVICE);
-
-        providers = GeoUtils.evaluateProviders(locationManager);
+        locationController = CommCareLocationControllerFactory.Companion.getLocationController(this, this);
```

Comment:
you can provide a `@JvmStatic` annotation for `getLocationController` to avoid using `Companion` in references.

Reply:
Yay! Thanks! I didn't know about this.